src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyState, action.extraData);
+
+        break;
+    }
+
+    switch (action.type) {
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete copyState[key];
+        }
+
+        break;
+    }
+
+    switch (action.type) {
+      case 'clear':
+        for (const key in copyState) {
+          delete copyState[key];
+        }
+    }
+
+    resultArr.push({ ...copyState });
+  }
+
+  return resultArr;
 }
```

Comment:
You don't need to write a switch every time. Just write it once and iterate over the cases. You can see the example [here](https://www.w3schools.com/js/js_switch.asp)
```
switch(expression) {
  case x:
    // code block
    break;
  case y:
    // code block
    break;
  default:
    // code block
}
```